src/Chapter4.hs

```diff
@@ -470,25 +485,42 @@ Applicatives can be found in many applications:
 
 Implement the Applicative instance for our 'Secret' data type from before.
 -}
+
 instance Applicative (Secret e) where
-    pure :: a -> Secret e a
-    pure = error "pure Secret: Not implemented!"
+  pure :: a -> Secret e a
+  pure = Reward
+
+  (<*>) :: Secret e (a -> b) -> Secret e a -> Secret e b
+  (Trap e) <*> _ = Trap e
+  (Reward e) <*> f = fmap e f -- why not e f 
+                              -- fmap wants function as first arg?
 
-    (<*>) :: Secret e (a -> b) -> Secret e a -> Secret e b
-    (<*>) = error "(<*>) Secret: Not implemented!"
 
 {- |
 =⚔️= Task 5
 
 Implement the 'Applicative' instance for our 'List' type.
-
+:
 🕯 HINT: in the applicative instance for lists, you have a list of
   functions and a list of arguments for those functions. You need to
   apply each function to each argument and combine all the results. You
   may also need to implement a few useful helper functions for our List
   type.
 -}
 
+-- helper function b/c ++ is missing on List
+append :: List a -> List a -> List a
+append Empty l2 = l2
+append l1 Empty = l1
+append (Cons e a) l2 = Cons e (append a l2)
+
+instance Applicative List where
+  pure :: a -> List a
+  pure _ = Empty
```

Comment:
Here you will lose the information on the provided value. You need to use`a` in here as well as it would be much more helpful type if it has some values rather than just `Empty`.